src/PrivateKey.cpp

```diff
@@ -35,6 +36,11 @@ PublicKey PrivateKey::getPublicKey(PublicKeyType type) const {
         result[0] = 1;
         ed25519_publickey(bytes.data(), result.data() + 1);
         break;
+    case PublicKeyType::ed25519Blake2b:
+        result.resize(PublicKey::ed25519Size);
+        result[0] = 11;
+        ed25519_publickey_blake2b(bytes.data(), result.data() + 1);
```

Comment:
I haven't looked into ed25519, but this looks weird to me. Why is the hash part of the function name? Can it be a parameter?

Reply:
It's the way the underlying ed25519-donna library implements customising the hash function used.

Reply:
Yes, but sometimes we need to question the libraries we use and whether they are the best approach. I'm not suggesting to rewrite, but perhaps modify? It looks like we could pass a struct with the function pointers to the hash function instead of using macros.

Reply:
In my opinion the trezor-core chosen approach and what you're suggesting are pretty much identical form a consumer perspective (passing in a struct-pointer/enum vs choosing the right function/function-pointer). What modifying trezor-core/ed25519-donna would do, however, is increase the maintenance cost and also make it harder to pull in code from upstream trezor-core or other trezor-core forks.